source/docs/casper/resources/tutorials/advanced/cross-contract.md

```diff
@@ -0,0 +1,681 @@
+   Compiling contract v0.1.0 (/Users/karolmarter/Desktop/Rust_Projects/cross-contract-2/contract)
+    Finished release [optimized] target(s) in 0.69s
+wasm-strip contract/target/wasm32-unknown-unknown/release/contract.wasm 2>/dev/null | true
+```
+
+Be sure to copy the keys from the previous folder to the new contract folder. We will not be creating a new set of keys and instead reusing those, which we already have. The call from the terminal will look as follows:
+
+```bash
+casper-client put-deploy \
+    --node-address http://136.243.187.84:7777 \
+    --chain-name casper-test \
+    --secret-key ./keys/secret_key.pem \
+    --payment-amount 20000000000 \
+    --session-path ./contract/target/wasm32-unknown-unknown/release/contract.wasm
+```
+
+:::tip
+
+We can notice that the contract.wasm is always the same when creating a new `cargo casper` project. While this is not crucial for small projects try to name the contracts appropriately in the `Makefile` in the main directory. We encourage to look at it and experiment a little with what happens when the `.wasm` filename will be changed and the following commands are called:
+
+```bash
+make prepare
+make build-contract
+```
+:::
+
+After the deploy we can check if it was successful and control the parameters of the deployed entry points.
+
+The output of the deployment is:
+
+```bash
+{
+  "id": -7557689417621513622,
+  "jsonrpc": "2.0",
+  "result": {
+    "api_version": "1.4.13",
+    "deploy_hash": "faeb7e4f010c20c88d2dd126da545933c26fd8ce370282b8cd49f7f6fe7304b9"
+  }
+}
+```
+
+:::tip
+
+If the contract name doesn't change during concurrent deploys, the urefs/hashes will be overwritten in the accounts named keys.
+
+:::
+
+If we control the deploy we can see that it has been successful:
+
+<img class="align-center" src={useBaseUrl("/image/tutorials/cross-contract/Deploy.png")} width="600" alt="Deploy" />
+
+If we check the accounts named keys we can see all of the accounts deployed contracts:
+
+<img class="align-center" src={useBaseUrl("/image/tutorials/cross-contract/NamedKeys.png")} width="600" alt="NamedKeys" />
+
+As we have now managed to deploy two contracts, we can call the endpoint of this contract passing appropriate arguments to the function. Just remember that invoking an entry point of the contract is also a deploy to the network.
+
+Let us quickly check the state of the message in the first contract:
+
+<img class="align-center" src={useBaseUrl("/image/tutorials/cross-contract/HelloWorldBCC.png")} width="600" alt="HelloWorld" />
+
+This is a simple `hello world` string. After invoking the endpoint using the command below this value should change.
+
+:::info
+
+`--session-hash` - is the contract hash, which is called through the put-deploy. Be sure to put your contract hash here.
+
+`--session-arg` "hello world_contract:Key= ..." takes the hash of the contract which we want to call from within the session-hash contract.
+
+:::
```

Comment:
Maybe we could ad the screenshot with the named keys again here saying that both hashes can be found here. Or at least mention that that's the best place to find them.

Reply:
I think placing the screenshot of the named keys again would be redundant. 10 Lines above we already have a screenshot of those keys.